flytekit/clis/sdk_in_container/run.py

```diff
@@ -249,13 +249,23 @@ class RunLevelParams(PyFlyteParams):
     )
     cluster_pool: str = make_click_option_field(
         click.Option(
-            param_decls=["--cluster-pool", "cluster_pool"],
+            param_decls=["--cluster-pool"],
```

Comment:
All the other options support an underscore version and technically this is backwards breaking so I'd vote for reverting this change.

(We need to have another discussion outside of this PR whether the `--cluster-pool` arg needs to be removed entirely because I'm not sure this ever worked.)

Reply:
I'm not sure why `--cluster-pool` wouldn't work? My reason for removing the latter was the missing `--`. I figured it is a ripple effect where the previous ones didn't have the `--` and so it was missed in the subsequent ones. Is it ok to have a config without `--`?

Reply:
> I'm not sure why --cluster-pool wouldn't work?

Not related to this PR, I was referring to [this](https://github.com/flyteorg/flyte/pull/4956#issuecomment-2114843886) discussion.